src/main/java/com/connexta/search/common/Index.java

```diff
@@ -18,44 +18,66 @@
 import static com.connexta.search.common.configs.SolrConfiguration.TITLE_ATTRIBUTE;
 
 import java.util.Date;
+import javax.annotation.Nullable;
 import lombok.Builder;
 import lombok.EqualsAndHashCode;
+import lombok.Generated;
 import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
 import org.springframework.data.annotation.Id;
 import org.springframework.data.solr.core.mapping.Indexed;
 import org.springframework.data.solr.core.mapping.SolrDocument;
 
 @SolrDocument(collection = SOLR_COLLECTION)
-@Getter
+@Builder
 @EqualsAndHashCode
-@Builder(toBuilder = true)
+@Generated
+@Getter
+@Setter
```

Comment:
❓ Do we need the setter if we have the builder?

Reply:
I agree. I think Index should be immutable. See https://github.com/connexta/ion-search/pull/86#discussion_r350883675, https://github.com/connexta/ion-search/pull/86#discussion_r350886871, and https://github.com/connexta/ion-search/pull/86#discussion_r350901744.